pkg/generate/app/cmd/newapp.go

```diff
@@ -919,7 +973,21 @@ func (c *AppConfig) run(acceptors app.Acceptors) (*AppResult, error) {
 		return nil, err
 	}
 
-	if err := c.resolve(components); err != nil {
+	var imageComp app.ComponentReference
+	imageComp, repositories, err = c.addImageSource(repositories)
```

Comment:
I don't like this being after validate - why wouldn't this just be part of validate?